src/session_impl.cpp

```diff
@@ -5162,11 +5154,21 @@ namespace aux {
 		// proxy, and it's the same one as we're using for the tracker
 		// just tell the tracker the socks5 port we're listening on
 		if (m_socks_listen_socket && m_socks_listen_socket->is_open())
-			return m_socks_listen_socket->local_endpoint().port();
+		{
+			error_code ec;
+			tcp::endpoint endp = m_socks_listen_socket->local_endpoint(ec);
+			if (ec)
```

Comment:
I'm experimenting with `BOOST_NO_EXCEPTIONS`, but still using exceptions enabled at compiler level. I'm not really that familiar with boost and asio thinking in order to say something about "naked throw expressions" vs `boost::throw_exception`.

What do you mean with "that should be changed instead"? The `socket_type` API? Do you want the `BOOST_NO_EXCEPTIONS` guard to be removed there?